connector_elasticsearch/models/se_index.py

```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class SeIndex(models.Model):
+
+    _inherit = "se.index"
+
+    config_id = fields.Many2one(
```

Comment:
Regarding the way to manage the config, we have the same issue on Algolia.
In Algolia case we must define which field are indexed (url key, attribute for filtering, categories...)
Raph have started a PR (https://github.com/akretion/connector-search-engine/pull/37/files) to integrate a generic logic but before merging it I think it worth to take the time to understand what exactly elastic need and compare it to algolia and make a generic part in search engine.

Reply:
@sebastienbeau  In the case of elasticsearch the setting must be provided when we create the indexes. All we need is a json document.. The way and the format used to provide the settings will differ from one SE to another. I don't if we should take time to implement a generic approach based on only 2 implementations.

Reply:
@lmignon I would love to test and review your Elastic PRs, specially as in Brazil anything that is priced in euros/USD like Algolia can quickly be something expensive... If you could write some 20 lines on how to start hacking Shopinvader with Elastic, that would help (I'm comfortable with the normal/Algolia stack already)...

Reply:
@rvalyi As you can see we are working on this topis :smirk: and we've a working implementation for Odoo 10 AND Locomotive with the same level of functionalities as the one provided with Algolia.

The odoo part:
https://github.com/akretion/connector-search-engine/pull/48
https://github.com/akretion/odoo-shopinvader/pull/288

The locomotive part (POC) 
https://github.com/akretion/shopinvader-template/pull/10
https://github.com/akretion/locomotive-shopinvader/pull/30

We now finalize this work on the connector part on top of Odoo-12 before back-porting all this work into 10.0. (So we benefit of all the cleaning done by @simahawk 

All this work on Elasticsearch is  the result of investigations and developments made by @Laurent-Corron . In addition to the python part he also found and fully integrated the [searchkit library](https://github.com/searchkit/searchkit) into the shopinvader template and has adapted the ruby code.

Last but not least, he also enriched the demo data to be able to have a relatively complete demo site (using the content of https://demo.dev.shopinvader.com/) https://github.com/akretion/odoo-shopinvader/pull/287

We'll have a full working stack on Odoo 12 in the next weeks.